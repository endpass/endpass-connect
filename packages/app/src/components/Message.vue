<template>
  <p :class="{ message: true, error: error, ellipsis: ellipsis }">
    <slot/>
  </p>
</template>

<script>
export default {
  name: 'Message',

  props: {
    ellipsis: {
      type: Boolean,
      default: false
    },

    error: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="postcss">
.message {
  font-size: 1em;

  &.error {
    color: #f00;
  }

  &.ellipsis {
    width: 100%;
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  }
}
</style>
