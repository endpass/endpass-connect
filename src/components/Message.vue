<template>
  <p
    :class="{
      message: true,
      error: error,
      success: success,
      ellipsis: ellipsis,
    }"
  >
    <slot />
  </p>
</template>

<script>
export default {
  name: 'Message',

  props: {
    ellipsis: {
      type: Boolean,
      default: false,
    },

    error: {
      type: Boolean,
      default: false,
    },

    success: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="postcss">
.message {
  font-size: 1em;
  line-height: 1.5;

  &.error {
    color: #f00;
  }

  &.success {
    color: #4dce4d;
  }

  &.ellipsis {
    width: 100%;
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  }
}
</style>
