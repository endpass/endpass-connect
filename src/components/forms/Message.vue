<template>
  <div>
    <form-field>
      <message data-test="form-message">{{ message }}</message>
    </form-field>
    <form-controls>
      <v-button
        :disabled="!closable"
        data-test="cancel-button"
        @click="emitCancel"
        >Close</v-button
      >
    </form-controls>
  </div>
</template>

<script>
import VButton from '../VButton.vue';
import Message from '../Message.vue';
import FormField from '../FormField.vue';
import FormControls from '../FormControls.vue';

export default {
  name: 'MessageForm',

  props: {
    message: {
      type: String,
      required: true,
    },

    closable: {
      type: Boolean,
      default: false,
    },
  },

  methods: {
    emitCancel() {
      this.$emit('cancel');
    },
  },

  components: {
    VButton,
    Message,
    FormField,
    FormControls,
  },
};
</script>
