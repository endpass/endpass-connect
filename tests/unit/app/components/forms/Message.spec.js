import { shallowMount, mount } from '@vue/test-utils';
import Message from '@/components/forms/Message.vue';

describe('Message', () => {
  describe('render', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowMount(Message, {
        propsData: {
          message: 'foo',
        },
      });
    });

    it('should correctly render Message component', () => {
      expect(wrapper.name()).toBe('MessageForm');
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = mount(Message, {
        propsData: {
          message: 'foo',
        },
      });
    });

    describe('cancel feature', () => {
      it('should emit close on click cancel button by default', () => {
        wrapper.find('[data-test=cancel-button]').vm.$emit('click');

        expect(wrapper.emitted().cancel).toBeTruthy();
      });
    });
  });
});
