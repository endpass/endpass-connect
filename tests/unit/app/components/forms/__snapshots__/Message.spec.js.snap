// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message render should correctly render Message component 1`] = `
<div>
  <form-field-stub>
    <message-stub data-test="form-message">foo</message-stub>
  </form-field-stub>
  <form-controls-stub>
    <v-button-stub disabled="true" data-test="cancel-button">Close</v-button-stub>
  </form-controls-stub>
</div>
`;
